.gitignore

```diff
@@ -12,3 +12,6 @@ Cargo.lock
 *.swo
 *.swp
 *.vim
+
+# temporary folder to refresh SDK with TypeSpec
+TempTypeSpecFiles/
```

Comment:
Why is this in the repo at all? We should use a subdirectory under `$TEMP`, which is universal across platforms.

Reply:
this is current tsp-client behavior, if you really want a change for this behavior please open an issue for tsp-client here and we can discuss: https://github.com/Azure/azure-sdk-tools/issues

I dont think this should block this PR from merging though if you want to start using tsp-client to generate client libraries, it's an implementation detail of the tool.

Reply:
https://github.com/Azure/azure-sdk-tools/issues/9248